test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -861,6 +861,111 @@ public void GetCommaSeparatedValues_OutputsEmptyArrayForNullValue()
             Assert.Empty(result);
         }
 
+        public static TheoryData InvalidTagHelperAttributeDescriptorData
+        {
+            get
+            {
+                var errorFormat = "Invalid tag helper bound property '{0}' binding to HTML attribute " +
+                    "'{1}' for type '{2}'. Tag helpers cannot bind to attributes beginning with 'data-'.";
```

Comment:
great case where including the full error strings in the test data would improve understanding